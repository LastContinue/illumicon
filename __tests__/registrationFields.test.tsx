import * as React from 'react'
import RegistrationFields from '../components/registrationFields';
import renderer from 'react-test-renderer'

const islands = [
    { name: "Florio", type: "Park" },
    { name: "Voluca", type: "Desert" }
]

it('renders correctly', () => {
    const tree = renderer
        .create(<RegistrationFields islands={islands} />)
    expect(tree).toMatchSnapshot();
})