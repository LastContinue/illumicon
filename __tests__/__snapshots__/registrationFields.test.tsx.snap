// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className="mb-4"
  >
    <label
      className="block text-gray-700 text-sm font-bold mb-2"
      htmlFor="name"
    >
      Name
    </label>
    <input
      className="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
      id="name"
      maxLength={30}
      minLength={3}
      placeholder="Name"
      required={true}
      type="text"
    />
  </div>,
  <div
    className="mb-6"
  >
    <label
      className="block text-gray-700 text-sm font-bold mb-2"
      htmlFor="email"
    >
      Email
    </label>
    <input
      className="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
      id="email"
      maxLength={100}
      placeholder="professor@pokelab.com"
      required={true}
      type="email"
    />
  </div>,
  <div
    className="mb-6"
  >
    <label
      className="block text-gray-700 text-sm font-bold mb-2"
      htmlFor="tour"
    >
      Island Tour
    </label>
    <select
      className="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
      id="tour"
    >
      <option
        value="Florio"
      >
        Florio
      </option>
      <option
        value="Voluca"
      >
        Voluca
      </option>
    </select>
  </div>,
  <div
    className="flex items-center justify-between"
  >
    <button
      className="bg-green-500 hover:bg-green-700 text-white font-bold py-2 px-4 rounded focus:outline-none focus:shadow-outline"
      type="submit"
      value="Submit"
    >
      Register!
    </button>
  </div>,
]
`;
